various fixes: build on Linux etc.

ec1oud opened the pull request:
These fixes should be OK to merge I think.  However there's still something wrong with the usage of the network module: it's basically offline and resetting periodically (due to a watchdog?)  So, not sure if I'm still missing something in the Makefile which is necessary to get that working.

ec1oud committed:
add Makefile, and make it build on Linux

Filenames are case-sensitive

ec1oud committed:
fix some warnings

ec1oud committed:
can disable backlight timeout by setting to 0

ec1oud committed:
clean up trailing whitespace

My editor does this automatically, so I end up with lots of noisy diffs
if this isn't done in advance.

ec1oud committed:
LCD font: provide the missing last byte of the U character

ec1oud committed:
format dose with fixed width

This prevents wrapping the display incorrectly.

ec1oud committed:
BME280: don't zero-prefix the temperature or humidity

Having a space in case the temperature or humidity is less than
3 digits makes it easier to read.

ec1oud committed:
BME280: convert temperature to floating point accurately

fix from http://www.uradmonitor.com/topic/floating-point-temperature-in-bme280-cpp/

MalteP committed:
Adding PSTR macro to jsonKeyFind() call in callback_browser() needs _P string functions to be used here

MalteP committed:
Remove whitespaces

MalteP committed:
Include pgmspace.h as we are using _P string functions and PSTR macros

MalteP committed:
Fix missing humidity value on local webserver

MalteP committed:
Transmit tube type as suggested by Radu

ec1oud committed:
Merge pull request #1 from MalteP/master

Adding PSTR macro to jsonKeyFind() call needs _P string functions to be used here

MalteP committed:
Add whitespaces because of compiler warnings

ec1oud committed:
Merge pull request #2 from MalteP/master

Add whitespaces because of compiler warnings

ec1oud committed:
Makefile: add -Wl,-relax linker option

This makes the code a little smaller.  If I build with avr-gcc 4.8.2
on Ubuntu, it otherwise ends up just a little too large to fit in flash.

ec1oud committed:
pass all compile flags to the linker as well; generate a map file

http://stackoverflow.com/questions/20356871/avr-gcc-destructive-optimizations
says it's a good idea.

ec1oud committed:
strip unused functions to save space

http://stackoverflow.com/questions/6687630/how-to-remove-unused-c-c-symbols-with-gcc-and-ld

ec1oud committed:
scavenge some memory

even more PSTR; bit packing of some variables

ec1oud committed:
use avr-size to show memory usage after compilation

I want to add a ring buffer but memory is tight.

ec1oud committed:
use the preprocessor rather than switch statements to deal with tube type

A switch statement might take memory if the compiler generates a jump table:
http://thegaragelab.com/conserving-memory-on-an-avr/
(although it seems to me that it doesn't, with avr-gcc 4.8, at least).
The tube type will not change at runtime; we only need a multiplier
constant, and the name as a string for concatenation.  The compiler
can do string concatenation too, so there's no reason to use
printf("tube %s") to substitute a string which will never change.
This change saves 30 bytes of RAM.

ec1oud committed:
calculate some statistics: max, min, mean

For this we need a ring buffer to remember at least a few samples.
If we want to have a line graph later, we'll need to remember even more.
But more urgently the idea is to find the outliers, which requires
comparing against the mean... this isn't done yet.

ec1oud committed:
add stats to JSON response

cpmMin, cpmMax, cpmAverage

ec1oud committed:
convert RingBuf to a C++ template class

We'll soon need other buffers storing other types.

Wolferl1 commented:
I had the same problem with the network module being offline, when I first tried to compile the firmware on a Windows machine. You'll need to use at least sw version 123 to get something valid and working compiled. I couldn't be bothered to find out what the problem behind that was, though.

ec1oud committed:
send status out over serial port every 10 seconds

This makes possible logging in the field with an Adalogger or some such.

ec1oud committed:
update README; add programming setup pics

My memory sucks; if I hadn't left the programmer attached to the device,
I'd have forgotten how to connect it for programming, and that I even
had a 3.3V programmer in the first place, and what kind.

ec1oud committed:
add pic of battery slot
